aws/ec2/ec2/ebs.go

```diff
@@ -0,0 +1,81 @@
+package ec2
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/ec2"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type BlockDeviceArgs struct {
```

Comment:
Why going through all that for a dozen of fields?
Why not just having a simple function like:
```
func convertEbsToBlockDeviceArgs(ebs map[string]string) (*ec2.InstanceEbsBlockDeviceArgs, error) {
  args.Foo = ebs["bar"]
}
```